api/src/main/java/org/openmrs/Encounter.java

```diff
@@ -738,4 +738,22 @@ public Encounter copyAndAssignToAnotherPatient(Patient patient) {
 				.filter(o -> o.getOrderGroup() == null)
 				.collect(Collectors.toList());
 	}
+
+	/**
+	 * Check if encounter has a particular diagnosis
+	 *
+	 * @since 2.2
+	 * 
+	 * @param diagnosis the diagnosis to check if it belongs to this given encounter
+	 *                     
+	 * @return true if this encounter has the given diagnosis, else false
+	 */
+	public Boolean hasDiagnosis(Diagnosis diagnosis) {
+		for (Diagnosis diagnosisFound : getDiagnoses()) {
```

Comment:
Instead of diagnosisFound, i would rename it to just diagnosis because it is not yet found 😊

Reply:
Okay. Noted

Reply:
Since we already have diagnosis parameter, I'll name it checkedDiagnosis

Reply:
Just diagnosis is enough.

Reply:
Okay